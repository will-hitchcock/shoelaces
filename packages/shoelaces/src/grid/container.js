import query from './media.new';

const container = ({ fluid }) => {
  const props = fluid
    ? {
      width: '100%',
      paddingRight: 0,
      paddingLeft: 0,
      overflowX: 'hidden',
    }
    : query((width) => ({ width }))({
      sm: '34em', md: '45em', lg: '58em', xl: '70em', xx: '99em',
    });

  return {
    marginRight: 'auto',
    marginLeft: 'auto',
    ...props,
  };
};


export default container;
