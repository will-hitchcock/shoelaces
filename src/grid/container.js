import { flattenMin } from '../util/media'
import render from '../util/render'

const container = ({ fluid, ...props }) => {
  const width = fluid ?
  {
    width: '100%',
    paddingRight: 0,
    paddingLeft: 0,
    overflowX: 'hidden'
  } :
  flattenMin(
    { sm: '34em', md: '45em', lg: '58em', xl: '70em', xx: '99em' },
    width => ({ width })
  )

  return {
    marginRight: 'auto',
    marginLeft: 'auto',
    ...width
  }
}

export default render(container)
