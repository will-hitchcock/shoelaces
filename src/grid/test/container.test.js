import container from '../container'

describe('container', () => {
  it('renders default container styles', () => {
    expect(container()).toMatchSnapshot()
  })

  it('renders fluid container styles', () => {
    expect(container({ fluid: true })).toMatchSnapshot()
  })
})
