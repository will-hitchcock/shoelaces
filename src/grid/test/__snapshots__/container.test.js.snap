// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container renders default container styles 1`] = `
Object {
  "@media (min-width: 105em)": Object {
    "width": "99em",
  },
  "@media (min-width: 36em)": Object {
    "width": "34em",
  },
  "@media (min-width: 48em)": Object {
    "width": "45em",
  },
  "@media (min-width: 62em)": Object {
    "width": "58em",
  },
  "@media (min-width: 75em)": Object {
    "width": "70em",
  },
  "marginLeft": "auto",
  "marginRight": "auto",
}
`;

exports[`container renders fluid container styles 1`] = `
Object {
  "marginLeft": "auto",
  "marginRight": "auto",
  "overflowX": "hidden",
  "paddingLeft": 0,
  "paddingRight": 0,
  "width": "100%",
}
`;
